### ISSUE TITLE
NPE when calling setUserId/unsetUserId after app resume

### ISSUE THREAD

**Issue by sidnair** [SiftScience/sift-android#45]
Hi,

I'm running into an error when trying to use the latest version of the SDK:

```
04-12 11:24:03.020  6450  6450 E AndroidRuntime: Process: siftscience.android.hellosift, PID: 6450
04-12 11:24:03.020  6450  6450 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'void siftscience.android.SiftImpl.setUserId(java.lang.String)' on a null object reference
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at siftscience.android.Sift.setUserId(Sift.java:151)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at siftscience.android.hellosift.HelloSift.setUserId(HelloSift.java:69)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at siftscience.android.hellosift.HelloSift$1.onClick(HelloSift.java:45)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at android.view.View.performClick(View.java:6294)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at android.view.View$PerformClick.run(View.java:24770)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:790)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:164)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6494)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
04-12 11:24:03.020  6450  6450 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

I'm using a React Native app with top-level [application integration](https://github.com/SiftScience/sift-android#application-integration) + a simple native module to call setUserId/unsetUserId from JS. I can provide more context on my setup if that's helpful, but I was able to get a repro by modifying the HelloSift app in this repo, which will hopefully be more useful.

Here's what I did:
1. Edited repo so I could build and to call setUserId instead of opening OtherActivity when the button is tapped - see fork [here](https://github.com/sidnair/sift-android).
1. Install to device with `gradle installDebug`
1. Open app by tapping icon
1. Tap the "Set User Id" button and observe that there aren't errors (ignoring 401s due to unset creds)
1. Press the home button on Android to get out of the app
1. Open app by tapping icon
1. Press the "Set User Id" button and get a crash

It looks to me (from `adb logcat | grep -i sift`) like the data is getting archived when the app is paused but not unarchived on resume. Note that if instead of hitting the home button, I kill the app and re-open, it works, presumably because the open()/collect() get called and the data gets unarchived.

Please let me know if you need anything else from me or if doing something wrong. Thanks!

**Comment by quanpod** [comment]
Thanks for the report Sid, will have the team investigate. Appreciate the
detailed notes!

**Comment by garylee1** [comment]
@sidnair thanks for the report, and great catch – indeed, we're missing a call to `open` within `resume`. For now, you can either call that yourself instead of `resume`, or wait for the new release (hopefully I can push it out today!).

**Comment by garylee1** [comment]
@sidnair we just released [0.10.1-beta](https://github.com/SiftScience/sift-android/releases/tag/v0.10.1-beta) which fixes this!
